### PR TITLE
adding csl as an extension for XML, included sample

### PR DESCRIPTION
**Pull request by redroot**
Examples in the wild: https://github.com/search?utf8=%E2%9C%93&q=extension%3Acsl+NOT+nothack&type=Code&ref=searchresults

Documentation: http://docs.citationstyles.org/en/stable/specification.html#xml-declaration

Any questions or if I haven't added enough please let me now

### CODE REVIEW

**Comment by redroot** [comment]
I've accepted the license agreement but the tests still seem to be in failed state, any idea why?

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -3644,6 +3644,7 @@ XML:
   - .axml
   - .ccxml
   - .clixml
+  - .csl
```
Comment (pchaigno) [inline lib/linguist/languages.yml]: Extensions must be sorted (except for the first one) ;)

**Comment by redroot** [comment]
@pchaigno ah great spot, i've amended the original commit :+1:

**Comment by pchaigno** [comment]
@arfon Looks like the Travis build is not always shown in the web interface. Do you know why that might be?

**Comment by arfon** [comment]
> @arfon Looks like the Travis build is not always shown in the web interface. Do you know why that might be?

Yeah, I noticed this. I have no idea to be honest :-\ - I'll try and do some poking around to figure out what might be going on.

**Comment by redroot** [comment]
This build doesnt seem to have run at all as far as I can tell, are you guys seeing differently? https://travis-ci.org/github/linguist/pull_requests

**Comment by redroot** [comment]
Any reason this cant be merged?

**Comment by arfon** [comment]
@redroot - sorry for the delay. This change will be live in the next release of Linguist (next week sometime)

**Pull request merged**